src/custom/state/price/reducer.ts

```diff
@@ -115,17 +116,29 @@ export default createReducer(initialState, (builder) =>
     .addCase(updateQuote, (state, action) => {
       const quotes = state.quotes
       const payload = action.payload
-      const { sellToken, chainId } = payload
+      const { sellToken, chainId, isBestQuote } = payload
       initializeState(quotes, action)
 
       // Updates the new price
       const quoteInformation = quotes[chainId][sellToken]
-      if (quoteInformation) {
+      const quote = current(state).quotes[chainId]
+
+      // Flag to not update the quote when the there is already a quote price and the
+      // current quote in action is not the best quote, meaning the best quote for
+      // some reason was already loaded before fast quote and we want to keep best quote data
+      const shouldNotUpdate = quote && quote[sellToken]?.price?.amount && !isBestQuote
```

Comment:
Also, nitpick. Move the `!isBestQuote` to the first condition, so its evaluated first in short-circuit